manta-trusted-setup/src/groth16/ceremony/signature.rs

```diff
@@ -0,0 +1,149 @@
+// Copyright 2019-2022 Manta Network.
+// This file is part of manta-rs.
+//
+// manta-rs is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// manta-rs is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with manta-rs.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Groth16 Trusted Setup Ceremony Signatures
+
+use manta_crypto::signature;
+use manta_util::{serde::Serialize, AsBytes};
+
+/// Nonce
+pub trait Nonce: PartialEq {
+    /// Increments the current nonce by one.
+    fn increment(&self) -> Self;
+
+    /// Checks if the current nonce is valid.
+    fn is_valid(&self) -> bool;
+}
+
+impl Nonce for u64 {
+    #[inline]
+    fn increment(&self) -> Self {
+        self.saturating_add(1)
```

Comment:
Why `saturating_add`? Since `Self::MAX` nonce is invalid, we can do whatever we want here right (for example, `*self += 1`)?